test/features/composability/mandos-legacy/l_proxy_test_message_otherShard.scen.json

```diff
@@ -51,7 +51,8 @@
                     "nonce": "0",
                     "balance": "0",
                     "storage": {
-                        "str:other_contract": "sc:proxy-second"
+                        "str:other_contract": "sc:proxy-second",
+                        "str:ARWEN@ASYNC1...............................": "0x0a20000000000000000070726f78792d66697273745f5f5f5f5f5f5f5f5f5f5f5f5f1220312e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2e2220615f757365725f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f5f60016801"
```

Comment:
Currently, the callback will fail, because this case is not handled. But this is indeed an issue and needs special handling. The VM can detect this situation though, it should be simple to implement a special path for this, guarded with an activation flag.

Reply:
add a TODO and a task to resolve this.